kurtosis-devnet/pkg/kurtosis/endpoints.go

```diff
@@ -63,187 +50,233 @@ func NewServiceFinder(services inspect.ServiceMap, opts ...ServiceFinderOption)
 		services:        services,
 		nodeServices:    []string{"cl", "el"},
 		l2ServicePrefix: "op-",
-		globalServices:  []string{"op-faucet"},
 	}
 	for _, opt := range opts {
 		opt(f)
 	}
+
+	f.triage()
 	return f
 }
 
-// FindL1Services finds L1 nodes.
-func (f *ServiceFinder) FindL1Services() ([]descriptors.Node, descriptors.RedundantServiceMap) {
-	return f.findRPCEndpoints(func(serviceName string) (string, int, bool) {
-		// Find node services and global services
-		allServices := append(f.nodeServices, f.globalServices...)
-		for _, service := range allServices {
-			if strings.HasPrefix(serviceName, service) {
-				// strip the L2 prefix if it's there.
-				name := strings.TrimPrefix(serviceName, f.l2ServicePrefix)
-				tag, idx := f.serviceTag(name)
-				return tag, idx, true
+type chainAcceptor func(*spec.ChainSpec) bool
+
+type serviceParser func(string) (int, chainAcceptor, bool)
+
+type triagedService struct {
+	tag    string // service tag
+	idx    int    // service index (for nodes)
```

Comment:
In near future I am thinking of redoing the L2 participants to look like challengers or superchains or supervisors:

```yaml
participants:
  node0:
    el:
      image: ...
    cl:
      image: ...
```

WDYT?

Reply:
sounds good to me. Hopefully we can introduce label-based metadata by then, so that it becomes transparent from here.